### PR TITLE
Search and sort_link added using ransack

### PR DESCRIPTION
**Pull request by coolRaider**
**What this PR does:**
- Added search form on the navbar
- Added sort_link for various article attributes

### CODE REVIEW

**Comment by coolRaider** [comment]
Will be adding tag search with Select2 integration.

File: app/assets/javascripts/tags.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (joney000) [inline app/assets/javascripts/tags.coffee:3]: remove comments/file

File: app/helpers/tags_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module TagsHelper
+end
```
Comment (joney000) [inline app/helpers/tags_helper.rb:2]: empty line at EOF

File: db/migrate/20160626053138_add_missing_taggable_index.acts_as_taggable_on_engine.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# This migration comes from acts_as_taggable_on_engine (originally 4)
+class AddMissingTaggableIndex < ActiveRecord::Migration
+  def self.up
+    add_index :taggings, [:taggable_id, :taggable_type, :context]
+  end
+
+  def self.down
+    remove_index :taggings, [:taggable_id, :taggable_type, :context]
+  end
+end
```
Comment (joney000) [inline db/migrate/20160626053138_add_missing_taggable_index.acts_as_taggable_on_engine.rb:10]: same, put empty line at EOF

**Comment by coolRaider** [comment]
these are auto-generated files. Do we need to edit these too?

**Comment by coolRaider** [comment]
Created new PR to merge into Tags branch.

**Comment by alienware** [comment]
Good going. Backmerge and close. 👍

**Pull request closed without merging**